src/main/java/org/apache/sling/feature/cpconverter/handlers/BundleEntryHandler.java

```diff
@@ -109,111 +154,309 @@ public void handle(@NotNull String path, @NotNull Archive archive, @NotNull Entr
+     * @throws Exception 
+     */
+    boolean extractSlingInitialContent(@NotNull JarEntry jarEntry, @NotNull InputStream bundleFileInputStream, @NotNull ArtifactId bundleArtifactId, @NotNull Collection<PathEntry> pathEntries, @NotNull Map<PackageType, VaultPackageAssembler> packageAssemblers, @NotNull JcrNamespaceRegistry nsRegistry, @NotNull ContentPackage2FeatureModelConverter converter) throws Exception {
+        final String entryName = jarEntry.getName();
+        // check if current JAR entry is initial content
+        Optional<PathEntry> pathEntry = pathEntries.stream().filter(p -> entryName.startsWith(p.getPath())).findFirst();
+        if (!pathEntry.isPresent()) {
+            return false;
+        }
+        Map.Entry<ContentParser, ParserOptions> contentParserAndOptions = getContentParserForEntry(jarEntry, pathEntry.get());
+
+        // https://sling.apache.org/documentation/bundles/content-loading-jcr-contentloader.html#file-name-escaping
+        String repositoryPath = (pathEntry.get().getTarget() != null ? pathEntry.get().getTarget() : "/") + URLDecoder.decode(entryName.substring(pathEntry.get().getPath().length()), "UTF-8");
+        String contentPackagePath = org.apache.jackrabbit.vault.util.Constants.ROOT_DIR + PlatformNameFormat.getPlatformPath(repositoryPath);
+
+        // in which content package should this end up?
+        VaultPackageAssembler packageAssembler = initPackageAssemblerForPath(bundleArtifactId, repositoryPath, pathEntry.get(), packageAssemblers, converter);
+        Path tmpInputFile = null;
+        if (contentParserAndOptions != null) {
+            // convert to docview xml
+            tmpInputFile = Files.createTempFile(converter.getTempDirectory().toPath(), "docview", ".xml");
+            try (OutputStream docViewOutput = Files.newOutputStream(tmpInputFile, StandardOpenOption.WRITE, StandardOpenOption.TRUNCATE_EXISTING);
+                 DocViewSerializerContentHandler contentHandler = new DocViewSerializerContentHandler(docViewOutput, nsRegistry)) {
+                contentParserAndOptions.getKey().parse(contentHandler, bundleFileInputStream, contentParserAndOptions.getValue());
+                contentPackagePath = FilenameUtils.removeExtension(contentPackagePath) + ".xml";
+            } catch (IOException e) {
+                throw new IOException("Can not parse " + jarEntry, e);
+            } catch (DocViewSerializerContentHandlerException e) {
+                throw new IOException("Can not convert " + jarEntry + " to enhanced DocView format", e);
+            }
+        }
 
-                Objects.requireNonNull(converter.getArtifactsDeployer()).deploy(new InputStreamArtifactWriter(input), id);
+        // does entry in initial content need to be extracted into feature model (e.g. for OSGi configurations)?
+        EntryHandler entryHandler = converter.getHandlersManager().getEntryHandlerByEntryPath(contentPackagePath);
```

Comment:
There is a bug here: the contentPackagePath doesn't start with a "/" - hence, it will never find a matching entryHandler for anything (as they assume the path starts with "/jcr_root/"). As I said above, I'm not sure it is the best idea to call the handlers manually at this point - it seems it would be much cleaner if we had a way to submit newly generated content package to the overall processing...

Anyways, I guess that means there are some tests missing that make sure we pick up content that should be handled by an entryHandler (like e.g. configurations in an install folder - which is one example where I found that a configuration wasn't processed).

Reply:
Do you have a real-life example of a bundle with Sling-Initial-Content also containing some OSGi configuration? I looked for that but couldn't find one....

Reply:
there is one in composum if you use this version: https://search.maven.org/remotecontent?filepath=com/composum/nodes/composum-nodes-aem-package/2.3.0/composum-nodes-aem-package-2.3.0.zip

```
[INFO] Created package com.composum.nodes:composum-nodes-config-apps:2.3.0 out of Sling-Initial-Content from 'com.composum.nodes:composum-nodes-config:2.3.0'
[INFO] Processing configuration 'org.apache.sling.jcr.base.internal.LoginAdminWhitelist.fragment~composum_core_v2'.
```

(Obviously, that was after I fixed the two issues)

Reply:
That the entry path starts with a `/` seems like a bug to me. IMHO same as for JAR/ZIP it should always be relative. The leading `/` comes from `BaseVaultPackageScanner#newPath` returning  / + `entryName` when `path` is empty. Should we fix this as well here (and in all handlers) or rather document that all entry paths are supposed to start with a `/`?

Reply:
I guess we can, but we have to fix all (or at least most) handlers then to make their patterns relative. Interestingly, that would mean the PrivilegesHandler would right now not match anything...

Reply:
Let us not do that change now. I just prepended the `/` and added some comment in https://github.com/apache/sling-org-apache-sling-feature-cpconverter/pull/74/commits/9a14efcb40289a6c14b1516cc9bb4f40fc187588

Reply:
@anchela is fixing the PrivilegesHandler in https://github.com/apache/sling-org-apache-sling-feature-cpconverter/pull/75